velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp

```diff
@@ -844,3 +844,39 @@ TEST_F(ParquetReaderTest, prefetchRowGroups) {
     }
   }
 }
+
+TEST_F(ParquetReaderTest, parseMapKeyValue) {
+  // Some existing data incorrectly used MAP_KEY_VALUE in place of MAP.
+  // For backward-compatibility, a group annotated with MAP_KEY_VALUE that is
+  // not contained by a MAP-annotated group should be handled as a MAP-annotated
+  // group. Details in
+  // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules-1
+  // Following is the schema in sample file
+  //  message hive_schema {
+  //    optional group test (MAP_KEY_VALUE) {
+  //      repeated group map {
+  //        required binary key (STRING);
+  //        optional int64 value;
+  //    }
+  //  }
+  const std::string filename("incorrect_map_key_value.parquet");
```

Comment:
I feel calling it "incorrect" might be misleading, it is actually just an older version right?  Marking it "incorrect" sounds we should expect the read to fail

Reply:
Let me update it.